kafka/logger.go

```diff
@@ -0,0 +1,45 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package kafka
+
+import (
+	"net"
+	"time"
+
+	"github.com/twmb/franz-go/pkg/kgo"
+	"go.uber.org/zap"
+)
+
+// Compile-time check that loggerHook implements the HookBrokerConnect interface.
+var _ kgo.HookBrokerConnect = new(loggerHook)
+
+type loggerHook struct {
+	logger *zap.Logger
+}
+
+// OnBrokerConnect implements the kgo.HookBrokerConnect interface.
+func (l *loggerHook) OnBrokerConnect(meta kgo.BrokerMetadata, dialDur time.Duration, _ net.Conn, err error) {
+	if err != nil {
+		l.logger.Error("failed to connect to broker",
+			zap.Error(err),
```

Comment:
Do you think it's also worthwhile to add `zap.Stack` to this log entry? There are multiple code paths that could call into connect/reconnect in franz-go, it would be nice to record them.

Reply:
Makes sense, thanks 👍🏼 
Hopefully the stacktrace is not generating too much output, since this is an Error level, it will be present in all failed connect attempts, and AFAIU from investigation made by @endorama we see a lot of them.
I'll add it, and we can remove later on if we see it's too verbose.